content/blog/CustomWebApiComponent/index.md

```diff
@@ -0,0 +1,384 @@
+
+Creating a new component is actually pretty simple and, as a benefit, you will also have the possibility to use Camel sub-projects to make your Web API available within different runtimes: [SpringBoot](https://github.com/apache/camel-spring-boot) (auto-config), [Quarkus](https://github.com/apache/camel-quarkus) (Java native compilation) and [Camel-K](https://github.com/apache/camel-k) (serverless work).
+
+This minimal, yet complete, example will focus on all the steps required to create and contribute your own Web API component to the Apache Camel codebase.
+
+As prerequisites you just need OpenJDK 1.8 and Maven 3.5+. 
+[Download the complete source code](camel-chuck.zip).
+
+## API endpoint
+
+To keep the example short and simple, we have chosen the [free JSON API for hand curated Chuck Norris facts](https://api.chucknorris.io). This is perfect to showcase how to create our new component, without having to deal with all the details and complexity of a full blown API.
+
+There are a few endpoints available, but we will use the following to retrieve a random joke in JSON format:
+
+```sh
+$ curl https://api.chucknorris.io/jokes/random | jq
+{
+  "categories": [],
+  "created_at": "2016-05-01 10:51:41.584544",
+  "icon_url": "https://assets.chucknorris.host/img/avatar/chuck-norris.png",
+  "id": "TMGDYHjLSu-G5Jkueph9lA",
+  "updated_at": "2016-05-01 10:51:41.584544",
+  "url": "https://api.chucknorris.io/jokes/TMGDYHjLSu-G5Jkueph9lA",
+  "value": "Who wins in a race car race? chuck Norris always wins. No excuses."
+}
+```
+
+## Checkout and initial setup
+
+The first step is to fork the official [Camel project on Github](https://github.com/apache/camel) and then clone it on your local machine with the following shell command (replace `$USERNAME` with yours):
+
+```sh
+$ git clone git@github.com:$USERNAME/camel.git
+$ cd camel/components
+```
+
+At this point we could use the `camel-archetype-component` to generate the project skeleton, but I usually prefer to do it from scratch:
+
+```sh
+$ mkdir -p camel-chuck/src/main/java/org/apache/camel/component/chuck \
+    && mkdir -p camel-chuck/src/main/docs/chuck-component.adoc \
+    && mkdir -p camel-chuck/src/test/java/org/apache/camel/component/chuck \
+    && mkdir -p camel-chuck/src/test/resources \
+    && cd camel-chuck
+```
+
+Now we need to create the project object module (POM) for Maven build (use the same parent version that you find in `camel/pom.xml`):
+
+```sh
+$ cat <<EOF > pom.xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
```

Comment:
We have a check for trailing whitespace, this is why the build failed. We just need the space at the end of this line removed.

```suggestion
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```